doc/port_auto_neg/port-auto-negotiation-design.md

```diff
@@ -396,9 +396,23 @@ To support validate interface speed on CLI side, a new field **supported_speeds*
 	; field                 = value
     ...
 	supported_speeds        = STRING                         ; supported speed list
+  speed                   = STRING                         ; operational speed
 
 An example value of supported_speeds could be "10000,25000,40000,100000".
 
+Before this feature, port speed in APP DB indicates both the configured speed and the operational speed. It is OK without this feature because port operational speed must be configured speed or port operational status is down. However, this is not true with this feature. Consider following flow:
+
+1. Configure port speed to 100G
+2. Configure advertised speed to 50G and enable auto negotiation
+3. Port operational speed turns to 50G
+4. Configure any port attribute, e.g mtu, portsyncd would put all port attributes to APP DB
```

Comment:
>all port attributes [](http://example.com/codeflow?start=62&length=19)

The right process should capture the changed attributes and only put the difference into APP DB. This could be achieved either by ConsumerStateTable or by the application.

Reply:
I agree. Should we put this logic to portmgrd? I suppose both portmgrd and portsyncd put port configuration to APP DB. I suppose portmgrd is newer and should be the right one, but I am not sure why portsyncd is still doing the job. Maybe is some kind of workaround before?

Reply:
portmgrd should subscribe ConfigDB port table and put the incremental attribute into AppDB.

Reply:
@Junchao-Mellanox agreed to fix this with another PR